src/driver-holtek/holtek8-shared.h

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: MIT
+/*
+ * Copyright © 2023 Michał Lubas
+ */
+
+#pragma once
+
+#include "driver-holtek.h"
+#include "libratbag-private.h"
+
+#define HOLTEK8_FEATURE_REPORT_SIZE 9
+#define HOLTEK8_BUTTON_SIZE 4
+#define HOLTEK8_MACRO_EVENT_SIZE 2
+#define HOLTEK8_MACRO_DATA_SIZE 128
+
+#define HOLTEK8_MAX_MACRO_EVENTS 256
+
+enum holtek8_api_version {
+	HOTLEK8_API_A = 1,
+	HOTLEK8_API_B = 2,
```

Comment:
> Unfortunately the ABI between devices is not stable but with supporting just 2 API types we are able to support a good fraction of devices.

If 'A' and 'B' are the names you chose yourself, add a comment that it's not some official names and they are just for libratbag, please, to avoid confusion

Reply:
Yes, these are completely arbitrary. Added a comment and a fun fact in the commit message.